modules/objc/examples/code_snippets/SampleCodeTest.m

```diff
@@ -1368,1143 +1376,1271 @@ - (void) dontTestListenerCertificateAuthenticatorCallback {
+    CBLReplicatorConfiguration* thisConfig; // TODO: Are we sure? this is Replicator config?
+    // tag::p2p-tlsid-tlsidentity-with-label[]
+    // tag::p2p-tlsid-check-keychain[]
+    // Check if Id exists in keychain and if so, use it
+    CBLTLSIdentity* identity = [CBLTLSIdentity identityWithLabel: @"doco-sync-server" error: &error]; // <.>
+
+    // end::p2p-tlsid-check-keychain[]
+    thisConfig.authenticator = [[CBLClientCertificateAuthenticator alloc] initWithIdentity: identity]; // <.>
+
+    // end::p2p-tlsid-tlsidentity-with-label[]
+    
+    // tag::p2p-tlsid-check-bundled[]
+    // CREATE IDENTITY FROM BUNDLED RESOURCE IF FOUND
+    
+    // Check for a resource bundle with required label to generate identity from
+    // return nil identify if not found
+    NSString* path = [[NSBundle mainBundle] pathForResource: @"doco-sync-server" ofType: @"p12"];
+    NSData* thisData = [NSData dataWithContentsOfFile: path
+                                              options: 0
+                                                error: &error];
+    if (!thisData)
+        return nil;
+    
+    // end::p2p-tlsid-check-bundled[]
+    
+    // tag::p2p-tlsid-import-from-bundled[]
+    // Use SecPKCS12Import to import the contents (identities and certificates)
+    // of the required resource bundle (PKCS #12 formatted blob).
+    //
+    // Set passphrase using kSecImportExportPassphrase.
+    // This passphrase should correspond to what was specified when .p12 file was created
+    CFArrayRef result = NULL;
+    NSDictionary* options = @{ (id)kSecImportExportPassphrase: @"couchbase" };
+    OSStatus status = SecPKCS12Import((__bridge CFDataRef)thisData, (__bridge CFDictionaryRef)options, &result);
+    if (status != errSecSuccess) {
+        NSLog(@"Failed to import data from provided with error :%d ", (int)status);
+        return nil;
+    }
+    
+    NSArray* importedItems = (NSArray*)CFBridgingRelease(result);
+    NSDictionary* item = importedItems[0];
+    
+    // Get SecIdentityRef representing the item's id
+    SecIdentityRef identityRef = (__bridge SecIdentityRef) item[(id)kSecImportItemIdentity];
+    
+    // Get Id's Private Key, return nil id if fails
+    SecKeyRef thisPrivateKey;
+    status = SecIdentityCopyPrivateKey(identityRef, &thisPrivateKey);
+    if (status != errSecSuccess) {
+        NSLog(@"Failed to import private key from provided with error :%d ", (int)status);
+        return nil;
+    }
+    CFAutorelease(thisPrivateKey);
+    
+    // Get all relevant certs [SecCertificate] from the ID's cert chain using kSecImportItemCertChain
+    NSArray* thisCertChain = item[(id)kSecImportItemCertChain];
+    
+    // Return nil Id if errors in key or cert chain at this stage
+    if (!thisCertChain || !thisPrivateKey)
+        return nil;
+    
+    // end::p2p-tlsid-import-from-bundled[]
+    
+    // tag::p2p-tlsid-store-in-keychain[]
+    // STORE THE IDENTITY AND ITS CERT CHAIN IN THE KEYCHAIN
```

Comment:
This seems to be out of scope. Also our CBLTLSIdentity automatically persists the certificate when using createIdentityForServer:attributes:expiration:label:error. Should this be change to use that method for creating the certificate and persist?